qutebrowser/browser/downloads.py

```diff
@@ -727,6 +736,23 @@ def _after_create_parent_question(self,
         else:
             self._after_set_filename()
 
+    def _get_conflicting_downloads(self):
+        """Returns a list of other active downloads with the same name."""
+        return [
+            download for download in
+            objreg.get('download-model', scope='window',
+                       window='current')._all_downloads()
+            if download is not self and
+            download._filename == self._filename and
+            not download.done
+        ]
+
+    def _cancel_conflicting_downloads(self):
+        """Cancel conflicting downloads and call _after_set_filename."""
+        for download in self._get_conflicting_downloads():
+            download.cancel()
```

Comment:
```suggestion
            download.cancel(remove_data=False)
```
If you're going to overwrite an existing file, don't remove the original file yet before the download has completed.

Reply:
Ah, good call. Thanks.